pkgs/test/default.nix

```diff
@@ -86,6 +86,25 @@ with pkgs;
   stdenv-inputs = callPackage ./stdenv-inputs { };
   stdenv = callPackage ./stdenv { };
 
+  # Unit tests for setup.sh
+  setup = import ./setup {
+    # NB: ^^^^^^
+    #     `import` and not `callPackage`, because we manipulate the whole `pkgs`
+    #     so that we can unit test without rebuilding everything; see setup/default.nix.
+    basePkgs =
+      if ! pkgs.config?baseCommit then pkgs
+      else import ../.. {
+        inherit (pkgs.stdenv.hostPlatform) system;
+        config = {
+          setupScript = "${
```

Comment:
```suggestion
          # Set the setup script to the version in the base commit
          # We need setup.sh to be a single file with the same store path as "${./setup.sh}",
          # so that stdenv equals stdenv for the base commit,
          # which turns out to be a bit complicated when it comes from a fetched source.
          setupScript = "${
```